src/WebJobs.Extensions.DurableTask/BasicLogWriter.cs

```diff
@@ -0,0 +1,134 @@
+    public class BasicLogWriter
+    {
+        private long fileSize;
+
+        /// <summary>
+        /// Simple log writer:
+        /// Rolls over at specified size (bytes)
+        /// Maximum of 2 files retained.
+        /// </summary>
+        /// <param name="logFolder">Directory where the logfile will be created.</param>
+        /// <param name="fileName">Log file name.</param>
+        /// <param name="rollOverSize">Maximum size of file (in bytes) before archiving it.</param>
+        public BasicLogWriter(string logFolder, string fileName, int rollOverSize)
+        {
+            this.LogFolder = logFolder;
+            this.RollOverSize = rollOverSize;
+            this.FileName = fileName;
+            this.Initialize();
+        }
+
+        internal long RollOverSize { get; private set; }
+
+        internal string LogFolder { get; private set; }
+
+        internal string FileName { get; private set; }
+
+        internal bool BufferingEnabled { get; set; }
+
+        private string LogFile
+        {
+            get
+            {
+                return Path.Combine(this.LogFolder, this.FileName);
+            }
+        }
+
+        private string LogFileBackup
+        {
+            get
+            {
+                return Path.Combine(this.LogFolder, this.FileName + ".1");
+            }
+        }
+
+        /// <summary>
+        /// Writes string argument to the log file.
+        /// </summary>
+        /// <param name="input">string to log.</param>
+        public void WriteLog(string input)
+        {
+            this.WriteDataInternal(input);
+        }
+
+        private void WriteDataInternal(string input)
+        {
+            try
+            {
+                // Roll-over if needed
+                if (this.fileSize >= this.RollOverSize)
+                {
+                    if (File.Exists(this.LogFileBackup))
+                    {
+                        try
+                        {
+                            File.Delete(this.LogFileBackup);
+                        }
+                        catch (Exception)
+                        {
+                        }
+                    }
+#if !FUNCTIONS_V1
+                    Syscall.rename(this.LogFile, this.LogFileBackup);
+                    this.fileSize = 0;
+#endif
+                }
+            }
+            catch (Exception)
+            {
+            }
+
+            try
+            {
+                byte[] dataToWrite = Encoding.UTF8.GetBytes(input);
+                this.fileSize += dataToWrite.Length;
+
+                // Write the data out
+                AppendAllBytes(this.LogFile, dataToWrite);
+            }
+            catch (Exception)
+            {
+            }
+        }
+
+        private static void AppendAllBytes(string fileName, byte[] data)
+        {
+            using (FileStream fileStream = new FileStream(fileName, File.Exists(fileName) ? FileMode.Append : FileMode.OpenOrCreate, FileAccess.Write))
```

Comment:
Non-blocking, but I wonder if we could reduce disk I/O by caching the `File.Exists` result, or checking if `this.fileSize > 0` instead of checking `File.Exists(fileName)`. I suppose this is more robust, though, in that it gracefully handles the case where someone manually deletes the log file while things are running, so maybe we should leave it as-is since our perf numbers are already good enough.

Reply:
Oh, maybe we're not using this anymore...